### ISSUE TITLE
Mario Kart 64 - various issues while using HD texture pack

### ISSUE THREAD

**Issue by Gagert** [gonetz/GLideN64#1286]
I'm currently using a HD texture pack for this game to make it appear more "modern". The one I use, is the texture pack by Kerber on this page http://textures.emulation64.com/index.php?id=downloads
(a typo on the linked front page however, nes_player4life is credited as author there instead of Kerber)

I have some issues with this texture pack however in this plugin; the icons on the main menu is missing the bottom part of the newer textures, as seen here:

![gliden64_mariokart64_000](https://cloud.githubusercontent.com/assets/8365446/20865205/474e6902-ba0c-11e6-938f-6a1ce8109f9f.png)

This is the very same image running with the Glide64 plugin:

![glide64_mariokart64_02](https://cloud.githubusercontent.com/assets/8365446/20865297/718afabc-ba0e-11e6-9f39-3db326fe3e94.png)

Also, it seems like GlideN64 would not render the main title screen like in Glide64; here's what GlideN64 outputs:

![gliden64_mariokart64_001](https://cloud.githubusercontent.com/assets/8365446/20865319/d9b8d46a-ba0e-11e6-82cf-03debd12b63b.png)

Here's the very same title screen with Glide64 as the main plugin:

![glide64_mariokart64_01](https://cloud.githubusercontent.com/assets/8365446/20865323/f7224806-ba0e-11e6-9f35-8d8163f8a03b.png)

Other than that, there seems to be some issues with the map in the Balloon Battle stages while using GlineN64. The map appears... white

![gliden64_mariokart64_000](https://cloud.githubusercontent.com/assets/8365446/20865341/3d9f7484-ba0f-11e6-96d0-15595640ba15.png)

The very same map screen in Glide64:

![glide64_mariokart64_03](https://cloud.githubusercontent.com/assets/8365446/20865344/50123b10-ba0f-11e6-8877-a1fef0c4b13f.png)

(I use a widescreen hack with Mario Kart 64, that's why the characters appears thin in the Glide64 plugin).

The last issue with the white map may not have anything to do with the texture pack in general, but more of a compatibility issue with GlideN64.

I'm using the GlideN64 Public Release 1.2 (or Pre-release as it says on the GitHub page of GlideN64).

Edit: Originally I wrote that nes_player4life is the author for the texture pack, while it's actually Kerber

**Comment by Nes-player4Life** [comment]
This pack has been plagued with issues. It shames me but Kerber's pack is the most complete pack to date.

**Comment by Gagert** [comment]
Oh, my mistake; I only read the description on the front page of the link that credited you as the author of the texture-pack. When clicking on the download link, I see Ambar Kerber credited as the actual author =)

**Comment by gonetz** [comment]
Hmm, looks like the same issue as #1341
Please check it again with build
https://drive.google.com/file/d/0B0YqMPjGo3B2N2R2Rll4NktXbWM/view?usp=sharing

**Comment by Gagert** [comment]
Looks like most of the HD textures are working in menus now. The battle stages however still have the same issue when it comes to only displaying a pure white minimap on the Battle Arenas (as seen in the picture at the beginning of this thread).

The Battle Arena minimaps display correctly while disabling texture packs in this plugin.

Most of all; great job gonatz, as always =)

**Comment by dezraj** [comment]
Thank you for reporting and fixing this issue, guys! 

By the way, is there a way to remove/avoid these black lines in the 3D models?

![mario kart 64 lines issue](https://cloud.githubusercontent.com/assets/27839354/25260113/64ec0020-2620-11e7-81b6-cb07f71572cb.jpg)

Thank you so much in advance for your time! :)

**Comment by fzurita** [comment]
Yes, enable 4xBRz texture enhancement

**Comment by DonelBueno** [comment]
@dezraj Enable the option "Render 2D elements in N64 resolution"

**Comment by dezraj** [comment]
@DonelBueno Enabling this option will fix these black lines in the 3D models but the course thumbnails in the "data" screen will become very blurry instead.
![mario kart 64 blurry courses](https://cloud.githubusercontent.com/assets/27839354/25300271/b5d91966-26e1-11e7-88c3-71b9bec5f41e.jpg)

I've also tried the "fix black lines between 2d elements" but it didn't fix the 3d models. So I guess I'll leave it like that for the moment. Thank you anyway! ;)

**Comment by Gagert** [comment]
Regarding [issue 1341](https://github.com/gonetz/GLideN64/issues/1341) which also covers the Mario Kart 64 Kerber texture-pack:

Using a newer WIP build of GlideN64 (the last one available as of today in the [WIP Builds 3 thread](https://github.com/gonetz/GLideN64/issues/1574)), most of the issues with the Kerber Mario Kart 64 texture-pack are gone (to my knowledge). 

One issue that still remains however is the minimap issue that is pictured on the top of this thread. Purely white minimaps on the Battle Arenas that's missing the transparency effect that can be viewed in the old Glide64 plugin.

I added this reply to this existing thread because of the last statement in [Issue 1341](https://github.com/gonetz/GLideN64/issues/1341): "I made several fixes in hires textures support. All issues should go. If something still not fixed, please open new issue."

**Comment by gonetz** [comment]
@Gagert I have no issues with minimap using Kerber's pack 2.5
![gliden64_mariokart64_000](https://user-images.githubusercontent.com/8983092/33470642-f5980d68-d69b-11e7-8e35-56cbba9b2191.png)

Be sure that "Use full transparencies" option is enabled on Texture pack tab.

**Comment by Gagert** [comment]
I managed to fix this now, by re-downloading the Kerber 2.5 pack and adding the Kerbs Fixed Pack in the same folder ([downloaded it here](http://www.emulation64.com/files/info/951/mario-kart-64-by-kerber2k.html/)).

Then I deleted the cached file of the Mario Kart 64 textures before starting Mario Kart 64 in Project 64 2.2.

Somehow the previous folder of the Kerber pack I had installed was corrupt; I may have messed up some filenames and/or some other things without remembering it now.

Well; I guess all the remaining bugs regarding GlideN64's handling of this good old texture-pack is now gone, to my knowledge.

(Personally I no longer use Project64 as my main N64 emulator, as I have migrated to RetroArch with the mupen64plus-libretro core. Someday those fixes would be integrated in the libretro-core I guess).
